core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -289,13 +290,23 @@ private[spark] class ExecutorAllocationManager(
       s" tasksperexecutor: $tasksPerExecutor")
     val maxNeeded = math.ceil(numRunningOrPendingTasks * executorAllocationRatio /
       tasksPerExecutor).toInt
-    if (tasksPerExecutor > 1 && maxNeeded == 1 && pendingSpeculative > 0) {
+    val totalNeed = if (tasksPerExecutor > 1 && maxNeeded == 1 && pendingSpeculative > 0) {
       // If we have pending speculative tasks and only need a single executor, allocate one more
       // to satisfy the locality requirements of speculation
       maxNeeded + 1
     } else {
       maxNeeded
     }
+
+    // Request additional executors to schedule the unschedulable tasks as well
+    if (numUnschedulables > 0) {
+      val maxNeededForUnschedulables = math.ceil(numUnschedulables * executorAllocationRatio /
+        tasksPerExecutor).toInt
+      math.max(totalNeed, executorMonitor.executorCountWithResourceProfile(rpId)) +
+        maxNeededForUnschedulables
+    } else {
```

Comment:
I think this should be:
```
math.max(totalNeed, executorMonitor.executorCountWithResourceProfile(rpId) + maxNeededForUnschedules)
```
If we were already going to request `maxNeededForUnschedulables` _more_ executors than we currently have as part of `totalNeed`, that already satisfies the requirements.

Reply:
Can you update with details about how this was resolved ? The original looked fine to me, but I want to make sure I am not missing something.

Reply:
Scratch that - what I had seen was probably a modified version, not original ...